spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala

```diff
@@ -60,6 +58,20 @@ class TableWriter[T] private (
     s"INSERT INTO ${quote(keyspaceName)}.${quote(tableName)} ($columnSpec) VALUES ($valueSpec) $ifNotExistsSpec$optionsSpec".trim
   }
 
+  private def deleteQueryTemplate(deleteColumns: ColumnSelector): String = {
+    val deleteColumnNames: Seq[String] = deleteColumns.selectFrom(tableDef).map(_.columnName)
+    val (primaryKey, regularColumns) = columns.partition(_.isPrimaryKeyColumn)
+    if (regularColumns.nonEmpty) {
+      logWarning(s"Only primary key column could be used in delete. ${regularColumns.mkString(", ")} columns are ignored")
```

Comment:
Maybe we should throw an error here, if regular columns are present tell the user to insert Nulls?